#!/bin/sh /usr/share/dpatch/dpatch-run
## 02-lockfile.dpatch by Jens Schmalzing <jensen@debian.org>
##
## DP: Adjusts location and content of the lockfile.

@DPATCH@

diff -Naur pxe-1.4.2.orig/autoconf.h.in pxe-1.4.2/autoconf.h.in
--- pxe-1.4.2.orig/autoconf.h.in	2002-08-22 18:06:11.000000000 +0000
+++ pxe-1.4.2/autoconf.h.in	2006-07-22 19:50:33.000000000 +0000
@@ -61,7 +61,7 @@
 /* should not need to change from here onwards */
 
 /* the lock file */
-#define LOCKFILE "/tmp/pxe.pid"
+#define LOCKFILE "/var/run/pxe/pxe.pid"
 
 #define DEF_MULTI_BOOT 0xe0000102
 #define DEF_ADDR "0.0.0.0"
diff -Naur pxe-1.4.2.orig/pxe.cc pxe-1.4.2/pxe.cc
--- pxe-1.4.2.orig/pxe.cc	2003-02-02 12:39:26.000000000 +0000
+++ pxe-1.4.2/pxe.cc	2006-07-22 19:50:33.000000000 +0000
@@ -312,7 +312,7 @@
 				umask(0);
 
 				// write out the pid
-				sprintf(pidnum, "%ld", (long)getpid());
+				sprintf(pidnum, "%ld\n", (long)getpid());
 				if(write(chk, pidnum, strlen(pidnum)) !=
 				  (ssize_t)strlen(pidnum)) {
 					std::cerr << "Unable to write lockfile\n";
